pkg/migrations/sql/preinstall/008-tracing.sql

```diff
@@ -0,0 +1,253 @@
+
+CREATE SCHEMA IF NOT EXISTS SCHEMA_TRACING;
+GRANT USAGE ON SCHEMA SCHEMA_TRACING TO prom_reader;
+
+CREATE SCHEMA IF NOT EXISTS SCHEMA_TRACING_PUBLIC;
+GRANT USAGE ON SCHEMA SCHEMA_TRACING_PUBLIC TO prom_reader;
+
+CREATE DOMAIN SCHEMA_TRACING_PUBLIC.trace_id uuid NOT NULL CHECK (value != '00000000-0000-0000-0000-000000000000');
+GRANT USAGE ON DOMAIN SCHEMA_TRACING_PUBLIC.trace_id TO prom_reader;
+
+CREATE DOMAIN SCHEMA_TRACING_PUBLIC.tag_k text NOT NULL CHECK (value != '');
+GRANT USAGE ON DOMAIN SCHEMA_TRACING_PUBLIC.tag_k TO prom_reader;
+
+CREATE DOMAIN SCHEMA_TRACING_PUBLIC.tag_v jsonb NOT NULL;
+GRANT USAGE ON DOMAIN SCHEMA_TRACING_PUBLIC.tag_v TO prom_reader;
+
+CREATE DOMAIN SCHEMA_TRACING_PUBLIC.tag_map jsonb NOT NULL DEFAULT '{}'::jsonb CHECK (jsonb_typeof(value) = 'object');
+GRANT USAGE ON DOMAIN SCHEMA_TRACING_PUBLIC.tag_map TO prom_reader;
+
+CREATE DOMAIN SCHEMA_TRACING_PUBLIC.tag_maps SCHEMA_TRACING_PUBLIC.tag_map[] NOT NULL;
```

Comment:
I originally had it named `tag_matchers`, but the fact that it was just an array of `tag_maps` made me change it. Do we name it after what it IS or how it's USED. :shrug:
I'll change it since we have a similarly named domain in the metrics stuff.